### ISSUE TITLE
Update documentation fo Unix Platforms

### ISSUE THREAD

**Issue by janhenke** [dotnet/coreclr#1590]
The documentation for building CoreCLR on Unix is outdated in some ways.

* Building CoreFX on Unix works now
* Building mscorlib on Unix works now (Linux instructions have been updated already, FreeBSD has not been updated)

It should also be considered if the instructions really should duplicate about building the managed assemblies. I think a reference to the corresponding instructions in the CoreFX repository are better (which would also avoid some of work here).

Last but not least, I think it would be a useful thing if we could merge the separate instructions for the different Unix platforms into one document. A lot of steps are more or less the same. It could work very well to have one document, which just outlines the differences between the platforms where needed. This would also reduce the maintenance burden regarding the documentation ;)

**Comment by janhenke** [comment]
@jasonwilliams200OK Maybe something for you?

**Comment by jasonwilliams200OK** [comment]
I agree with the consolidation of the Unices' instructions into one doc and then branch within where necessary.

My VM is bit untidy, as I built and installed Mono master for Roslyn. For the documentation purposes, it would be a good idea to start from scratch and validate each command/instruction at least for the lastest versions of CentOS, FreeBSD, OSX and Ubuntu. 

For example:

> there is no clang37 package.

this line in [current FreeBSD build instructions](https://github.com/dotnet/coreclr/blob/82a59ab92477689215fcc2822d8a298d1b6ff569/Documentation/building/freebsd-instructions.md) is untrue, because I get:

```sh
> pkg search clang37
clang37-3.7_2
> pkg search clang
clang-devel-3.8.d20150720_1
clang33-3.3_9
clang34-3.4.2_3
clang35-3.5.2
clang36-3.6.2
clang37-3.7_2
cloudabi-clang-3.7
```

Along the same lines, it would be really great if someone help us converting the bash script to sh, so we can avoid bash dependencies on platforms which comes packaged with csh, bourne, zsh etc. shells. Most part of the existing code work, except for commands like `source` etc. This is what node/npm and other libs devs prefer to make cross-unix scripts runnable with least common denominator: `/bin/sh`. 
We may inadvertently start supporting many other Unix flavors (Suse, Gentoo etc.) with this conversion alone. IMO, this bash->sh conversion is required in all related aspnet (https://github.com/aspnet/Home/issues/695) and dotnet repos.

In general, what I have gathered as best for cross-unix development is, always prefer least common denominator:
* POSIX system command first, if there is none, fallback to OS-specific.
* Use `sh` instead of some derived shell.

**Comment by arnmac** [comment]
I was working to validate and update the FreeBSD instructions. What needs to be done to build mscorelib on FreeBSD?

**Comment by jasonwilliams200OK** [comment]
@arnmac, mscorelib already builds on FreeBSD. You can grab the binary form CI build https://github.com/dotnet/coreclr/blob/master/README.md.

If you are interested in building from source yourself, follow the steps at https://github.com/dotnet/coreclr/blob/e9fa953/Documentation/building/freebsd-instructions.md and https://gist.github.com/jasonwilliams200OK/06db121ec534821fe541.
